code/modules/spells/spell_types/right_and_wrong.dm

```diff
@@ -1,8 +1,9 @@
 //In this file: Summon Magic/Summon Guns/Summon Events
 //and corresponding datum controller for them
 
-GLOBAL_DATUM(summon_guns, /datum/summon_things_controller)
-GLOBAL_DATUM(summon_magic, /datum/summon_things_controller)
+GLOBAL_DATUM(summon_guns, /datum/summon_things_controller/item)
+GLOBAL_DATUM(summon_magic, /datum/summon_things_controller/item)
+GLOBAL_DATUM(mass_teaching, /datum/summon_things_controller/spellbook_entry)
```

Comment:
Could you autodoc the global datum? And while you're here, also summon_guns and summon_magic :3